synapse/storage/database.py

```diff
@@ -2112,11 +2112,15 @@ def simple_delete_txn(
 
         Args:
             table: string giving the table name
-            keyvalues: dict of column names and values to select the row with
+            keyvalues: dict of column names and values to select the row with. If empty,
+                no rows will be deleted.
 
         Returns:
             The number of deleted rows.
         """
+        if not keyvalues:
+            return 0
```

Comment:
Sanity check: do we ever call this method with an empty set of key values?

I wonder if this ought to raise some kind of exception telling the author to use `simple_truncate`?

Reply:
It looks like we don't currently, but considering it's a no-op, it would likely aid in debugging if this did fail loudly. Good suggestion! I'll have it raise a `ValueError` and update the docstring.

Reply:
Done in 90128dd35.